browser/net/brave_oauth_protocol_handler.cc

```diff
@@ -0,0 +1,232 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/net/brave_oauth_protocol_handler.h"
+
+#include <string>
+#include <map>
+#include <utility>
+#include <vector>
+
+#include "base/optional.h"
+#include "base/strings/strcat.h"
+#include "base/strings/string_util.h"
+#include "base/task/post_task.h"
+#include "brave/common/url_constants.h"
+#include "brave/components/binance/browser/buildflags/buildflags.h"
+#include "brave/components/brave_rewards/browser/buildflags/buildflags.h"
+#include "brave/components/gemini/browser/buildflags/buildflags.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/web_contents.h"
+#include "net/base/escape.h"
+#include "net/base/url_util.h"
+#include "ui/base/page_transition_types.h"
+#include "url/gurl.h"
+
+#if BUILDFLAG(BINANCE_ENABLED)
+#include "brave/browser/binance/binance_service_factory.h"
+#include "brave/components/binance/browser/binance_service.h"
+#endif
+
+#if BUILDFLAG(BRAVE_REWARDS_ENABLED)
+#include "brave/components/brave_rewards/common/url_constants.h"
+#endif
+
+#if BUILDFLAG(GEMINI_ENABLED)
+#include "brave/browser/gemini/gemini_service_factory.h"
+#include "brave/components/gemini/browser/gemini_service.h"
+#endif
+
+namespace {
+
+const std::map<std::string, std::string> redirects = {
```

Comment:
non-trivial global vars are banned by CC (https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables), please use `NoDestructor` wrapper - there are many examples in the code

Reply:
prefer `base::flat_map` to `std::map`